### PR TITLE
Secrets generator redo

### PR DESCRIPTION
**Pull request by Dantemss**
LN says we shouldn't load Rails to create a file required for Rails to work so...

- Removed rails g secrets
- Added rake task that creates secrets.yml from example file and runs automatically before all rake tasks
- Changed secrets.yml format to nest related parameters, so everyone will need to recreate their secrets.yml

### CODE REVIEW

**Comment by Dantemss** [comment]
Will need changes to deployment to account for the lack of rails g secrets and the new secrets.yml format.

File: config/initializers/openstax_exchange.rb [hunk 1]
```diff
@@ -1,19 +1,21 @@
+secrets = Rails.application.secrets['openstax']['accounts']
```
Comment (lakshmivyas) [inline config/initializers/openstax_exchange.rb]: should this be ['openstax']['exchange']?

File: config/environments/production.rb [hunk 2]
```diff
@@ -85,18 +84,6 @@
...
-  if Rails.application.secrets[:fake_exchange]
-    # Don't try to connect to Exchange
-    OpenStax::Exchange.use_fake_client
-    OpenStax::Exchange.reset!
-  end
-
-  if Rails.application.secrets[:fake_exercises]
-    # Don't try to connect to Exercises
-    OpenStax::Exercises::V1.use_fake_client
-  end
```
Comment (lakshmivyas) [inline config/environments/production.rb]: No need to use fake client anymore?

Comment (lakshmivyas) [inline config/environments/production.rb]: nvm.  I see that there is a `stub` property below.

File: config/initializers/openstax_exchange.rb (hunk 1 above)
Comment (Dantemss) [inline config/initializers/openstax_exchange.rb]: Thanks, LN. Changed.

**Pull request merged**